File: core/src/com/dpashko/sandbox/scene/BaseScene.java
package com.dpashko.sandbox.scene;

import com.badlogic.gdx.Gdx;
import com.badlogic.gdx.graphics.g2d.SpriteBatch;
import com.dpashko.sandbox.font.FontsProvider;

public abstract class BaseScene<T extends BaseSceneController> {

    private final SpriteBatch batch;
    private final T controller;

    protected BaseScene(final T controller) {
        this.controller = controller;
        batch = new SpriteBatch();
    }

    public void render() {
        controller.tick();
        batch.begin();
        FontsProvider.defaultFont.draw(batch, "FPS: " + Gdx.graphics.getFramesPerSecond(), 5, Gdx.graphics.getHeight() - 10);
        batch.end();
    }

    public abstract void dispose();
}

File: core/src/com/dpashko/sandbox/scene/BaseScene.kt
package com.dpashko.sandbox.scene

import com.badlogic.gdx.Gdx
import com.badlogic.gdx.graphics.g2d.SpriteBatch
import com.dpashko.sandbox.font.FontsProvider

abstract class BaseScene<T : BaseSceneController> protected constructor(protected val controller: T) {

    private val batch: SpriteBatch = SpriteBatch()

    fun render() {
        controller.tick()
        batch.begin()
        FontsProvider.defaultFont.draw(batch, "FPS: ${Gdx.graphics.framesPerSecond}", 5f, Gdx.graphics.height - 10.toFloat())
        FontsProvider.defaultFont.draw(batch, "Heap: ${Gdx.app.javaHeap / 1_048_576L} mb.", 75f, Gdx.graphics.height - 10.toFloat())
        batch.end()
    }

    abstract fun dispose()

}